sourcetracker/tests/test_sourcetracker.py

```diff
@@ -10,119 +10,156 @@
+
+        _ftable = ftable.copy()
+
+        # A table with nans, expect a ValueError.
+        _ftable.iloc[0, 3] = np.nan
+        self.assertRaises(ValueError, check_and_correct_data, _ftable, True)
+
+        # Check that fractional counts give an error if function isn't allowed
+        # to correct them.
+        _ftable.iloc[0, 3] = 4.512
+        self.assertRaises(ValueError, check_and_correct_data, _ftable, False)
+
+        # Check that fractional counts are corrected.
+        ftable.iloc[0, 3] = 5
+        _ftable.iloc[0, 3] = 4.512
+        obs = check_and_correct_data(_ftable, True)
+        pd.util.testing.assert_frame_equal(ftable, obs)
+
+    def test_biom_to_df(self):
+        data = np.arange(200).reshape(20, 10).astype(float)
+        data2 = data.copy()
+        oids = ['o%s' % i for i in range(20)]
+        sids = ['s%s' % i for i in range(10)]
 
+        obs = biom_to_df(Table(data, oids, sids))
+        exp = pd.DataFrame(data.astype(np.int64).T, index=sids, columns=oids)
+        pd.util.testing.assert_frame_equal(obs, exp)
+
+        # Check that non integer data can be handled.
+        data[0, 3] = 4.512
+        obs = biom_to_df(Table(data, oids, sids))
+
+        data2[0, 3] = 5
+        exp = pd.DataFrame(data2.astype(np.int64).T, index=sids, columns=oids)
+        pd.util.testing.assert_frame_equal(obs, exp)
+
+    def test_remove_unshared_samples_and_order(self):
+        # Test an example where there are unshared samples present in both
+        # feature and sample tables. Notice that order is different between
+        # the samples that are shared between both tables. The order of samples
+        # in the returned tables is set by the ordering done in np.intersect1d.
+        sdata_c1 = [3.1, 'red', 5]
+        sdata_c2 = [3.6, 'yellow', 7]
+        sdata_c3 = [3.9, 'yellow', -2]
+        sdata_c4 = [2.5, 'red', 5]
+        sdata_c5 = [6.7, 'blue', 10]
+        samples = ['s1', 's4', 's2', 's3', 'sX']
+        headers = ['pH', 'color', 'day']
+        stable = pd.DataFrame([sdata_c1, sdata_c4, sdata_c2, sdata_c3,
+                               sdata_c5], index=samples, columns=headers)
+
+        fdata = np.arange(90).reshape(9, 10)
+        samples = ['s%i' % i for i in range(3, 12)]
+        columns = ['o%i' % i for i in range(1, 11)]
+        ftable = pd.DataFrame(fdata, index=samples, columns=columns)
+
+        exp_ftable = pd.DataFrame(fdata[[1, 0], :], index=['s4', 's3'],
+                                  columns=columns)
+        exp_stable = pd.DataFrame([sdata_c4, sdata_c3], index=['s4', 's3'],
+                                  columns=headers)
+
+        obs_stable, obs_ftable = remove_unshared_samples_and_order(stable,
+                                                                   ftable)
+
+        pd.util.testing.assert_frame_equal(obs_stable, exp_stable)
+        pd.util.testing.assert_frame_equal(obs_ftable, exp_ftable)
+
+        # No shared samples, expect a ValueError.
```

Comment:
Related to my above comment, it would make sense for each of these blocks to be implemented as a different function. In addition to the reason I mentioned above, it's helpful when you're debugging code as you can get better descriptions of what has failed.